src/Chapter2.hs

```diff
@@ -842,7 +862,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x l
+  | x < 0 = []
+  | otherwise = take (length l) $ drop x (cycle l)
```

Comment:
You are right! I think this would optimize this a bit:
```
rotate :: Int -> [a] -> [a]
rotate x l
  | x < 0 = []
  | otherwise = take lenL $ drop x' (cycle l)
  where
    lenL = length l
    x' = mod x lenL
```

Reply:
Exactly, using `mod` helps 🙂